core/vm/interpreter.go

```diff
@@ -124,6 +124,8 @@ func NewEVMInterpreter(evm *EVM, cfg Config) *EVMInterpreter {
 	}
 }
 
+var returnDataCopy string
```

Comment:
This is the hack that bypasses an apparent bug

Reply:
Curious, what was the bug - also why do we need to do this manually? Is`in.returnData` not always set correctly?

Reply:
This is a pretty crazy bug 😮

Has to do with internal Geth memory management... we gotta look into it further

Reply:
@gakonst unclear what the bug is. It appears that some combination of `RETURNDATACOPY` and other opcodes results in `MSTORE` overwriting `in.returnData`. Very strange.